TCAT/Controllers/HomeMapViewController.swift

```diff
@@ -0,0 +1,251 @@
+import MapKit
+import NotificationBannerSwift
+import SnapKit
+import UIKit
+
+protocol HomeMapViewDelegate {
+    func reachabilityChanged(connection: Reachability.Connection)
+}
+
+class HomeMapViewController: UIViewController {
+    
+    let userDefaults = UserDefaults.standard
+
+    var currentLocation: CLLocation?
+    var mapView: GMSMapView!
+    var loadingView = UIView()
+    var bounds = GMSCoordinateBounds()
+    var optionsCardVC: HomeOptionsCardViewController!
+    var delegate: HomeMapViewDelegate?
+    let reachability = Reachability(hostname: Network.ipAddress)
+    var locationManager = CLLocationManager()
+    var banner: StatusBarNotificationBanner? {
+        didSet {
+            setNeedsStatusBarAppearanceUpdate()
+        }
+    }
+    static let optionsCardInset = UIEdgeInsets.init(top: UIScreen.main.bounds.height/10, left: 20, bottom: 0, right: 20)
+    let minZoom: Float = 12
+    let defaultZoom: Float = 15.5
+    let maxZoom: Float = 25
+    let loadingIndicatorSize = CGSize.init(width: 40, height: 40)
+    
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return banner != nil ? .lightContent : .default
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupMapView()
+        setupOptionsCard()
+
+        setupConstraints()
+    }
+    
+    @objc func reachabilityChanged(_ notification: Notification) {
+        guard let reachability = notification.object as? Reachability else {
+            return
+        }
+        
+        // Dismiss current banner or loading indicator, if any
+        banner?.dismiss()
+        banner = nil
+        
+        delegate?.reachabilityChanged(connection: reachability.connection)
+        
+        switch reachability.connection {
+        case .none:
+            banner = StatusBarNotificationBanner(title: Constants.Banner.noInternetConnection, style: .danger)
+            banner?.autoDismiss = false
+            banner?.show(queuePosition: .front, on: navigationController)
+        default: break
+        }
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        navigationController?.isNavigationBarHidden = true
+        
+        // Add Notification Observers
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(reachabilityChanged(_:)),
+                                               name: .reachabilityChanged,
+                                               object: reachability)
+        do {
+            try reachability?.startNotifier()
+        } catch {
+            print("HomeVC viewDidLayoutSubviews: Could not start reachability notifier")
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        locationManager.delegate = self
+        locationManager.requestWhenInUseAuthorization()
+        
+        StoreReviewHelper.checkAndAskForReview()
+        
```

Comment:
do you know how conscious I am now of any goddamn n I t p I c









spacing